Index: machine.h
===================================================================
RCS file: /cvs/DragonFly/src/contrib/top/machine.h,v
retrieving revision 1.2
diff -u -p -u -r1.2 machine.h
--- machine.h	17 Jun 2003 04:24:07 -0000	1.2
+++ machine.h	4 Jun 2004 11:46:13 -0000
@@ -60,6 +60,8 @@ struct process_select
     int idle;		/* show idle processes */
     int self;		/* show self */
     int system;		/* show system processes */
+    int threads;	/* show threads */
+	int only_threads;	/* show only threads */
     int uid;		/* only this uid (unless uid == -1) */
     char *command;	/* only this command (unless == NULL) */
 };
