Index: top.c
===================================================================
RCS file: /cvs/DragonFly/src/contrib/top/top.c,v
retrieving revision 1.2
diff -u -p -u -r1.2 top.c
--- top.c	17 Jun 2003 04:24:07 -0000	1.2
+++ top.c	4 Jun 2004 12:09:22 -0000
@@ -193,9 +193,9 @@ char *argv[];
     fd_set readfds;
 
 #ifdef ORDER
-    static char command_chars[] = "\f qh?en#sdkriIuto";
+    static char command_chars[] = "\f qh?en#sdkriIutTOSo";
 #else
-    static char command_chars[] = "\f qh?en#sdkriIut";
+    static char command_chars[] = "\f qh?en#sdkriIutTOSO";
 #endif
 /* these defines enumerate the "strchr"s of the commands in command_chars */
 #define CMD_redraw	0
@@ -215,8 +215,11 @@ char *argv[];
 #define CMD_idletog2    13
 #define CMD_user	14
 #define CMD_selftog	15
+#define CMD_threads	16
+#define CMD_othreads	17
+#define CMD_system	18
 #ifdef ORDER
-#define CMD_order       16
+#define CMD_order       19
 #endif
 
     /* set the buffer for stdout */
@@ -245,6 +248,8 @@ char *argv[];
     ps.idle    = Yes;
     ps.self    = -1;
     ps.system  = No;
+    ps.threads = No;
+	ps.only_threads = No;
     ps.uid     = -1;
     ps.command = NULL;
 
@@ -271,7 +276,7 @@ char *argv[];
 	    optind = 1;
 	}
 
-	while ((i = getopt(ac, av, "SIbinquvs:d:U:o:t")) != EOF)
+	while ((i = getopt(ac, av, "SITONbinquvs:d:U:o:t")) != EOF)
 	{
 	    switch(i)
 	    {
@@ -301,6 +306,14 @@ char *argv[];
 		ps.idle = !ps.idle;
 		break;
 
+		  case 'O':
+		ps.only_threads = !ps.only_threads; /* only threads */
+		break;
+	
+	      case 'T':
+		ps.threads = !ps.threads;	/* show threads */
+	    break;
+
 	      case 'i':			/* go interactive regardless */
 		interactive = Yes;
 		break;
@@ -928,6 +941,31 @@ restart:
 				putchar('\r');
 				break;
 
+			    case CMD_threads:
+			        ps.threads = !ps.threads;
+			        new_message(MT_standout | MT_delayed,
+				   " %sisplaying threads.",
+				   ps.threads ? "D" : "Not d");
+				putchar('\r');
+				break;
+
+				case CMD_othreads:
+					ps.only_threads = !ps.only_threads;
+					new_message(MT_standout | MT_delayed,
+					ps.only_threads ?
+					  "Only displaying threads." :
+					  "Displaying threads and processes.");
+				putchar('\r');
+				break;
+
+			    case CMD_system:
+			        ps.system = !ps.system;
+			        new_message(MT_standout | MT_delayed,
+				   " %sisplaying system processes.",
+				   ps.system ? "D" : "Not d");
+				putchar('\r');
+				break;
+
 			    case CMD_user:
 				new_message(MT_standout,
 				    "Username to show: ");
