diff --git a/contrib/awk20100523/run.c b/contrib/awk20100523/run.c
index 88e0f8f..8db0af3 100644
--- a/contrib/awk20100523/run.c
+++ b/contrib/awk20100523/run.c
@@ -1152,13 +1152,13 @@ Cell *cat(Node **a, int q)	/* a[0] cat a[1] */
 	getsval(x);
 	getsval(y);
 	n1 = strlen(x->sval);
-	n2 = strlen(y->sval);
-	s = (char *) malloc(n1 + n2 + 1);
+	n2 = strlen(y->sval) + 1;
+	s = (char *) malloc(n1 + n2);
 	if (s == NULL)
 		FATAL("out of space concatenating %.15s... and %.15s...",
 			x->sval, y->sval);
-	strcpy(s, x->sval);
-	strcpy(s+n1, y->sval);
+	memmove(s, x->sval, n1);
+	memmove(s+n1, y->sval, n2);
 	tempfree(x);
 	tempfree(y);
 	z = gettemp();
