diff --git a/sys/contrib/dev/acpica-unix/osunixxf.c b/sys/contrib/dev/acpica-unix/osunixxf.c
index 967d544..44162e5 100644
--- a/sys/contrib/dev/acpica-unix/osunixxf.c
+++ b/sys/contrib/dev/acpica-unix/osunixxf.c
@@ -400,6 +400,7 @@ AcpiOsGetLine (
     UINT32                  *BytesRead)
 {
     UINT8                   Temp;
+    int                     c;
     UINT32                  i;
 
 
@@ -410,7 +411,13 @@ AcpiOsGetLine (
             return (AE_BUFFER_OVERFLOW);
         }
 
-        scanf ("%1c", &Temp);
+        c = scanf ("%1c", &Temp);
+        if (c != 1)
+        {
+                AcpiGbl_DbTerminateThreads = TRUE;
+                break;
+        }
+
         if (!Temp || Temp == '\n')
         {
             break;
