--- nroff.sh.orig	2010-01-26 08:02:39 +0100
+++ nroff.sh	2010-01-26 08:07:29 +0100
@@ -34,19 +34,23 @@
 case "`exec 2>/dev/null ; locale charmap`" in
   UTF-8)
     Tloc=utf8 ;;
-  ISO-8859-1 | ISO-8859-15)
+  ISO*8859-1 | ISO*8859-15)
     Tloc=latin1 ;;
   IBM-1047)
     Tloc=cp1047 ;;
+  KOI8-R)
+    T=-Tkoi8-r ;;
   *)
     case "${LC_ALL-${LC_CTYPE-${LANG}}}" in
       *.UTF-8)
         Tloc=utf8 ;;
-      iso_8859_1 | *.ISO-8859-1 | *.ISO8859-1 | \
-      iso_8859_15 | *.ISO-8859-15 | *.ISO8859-15)
+      iso_8859_1 | *.ISO*8859-1 | *.ISO8859-1 | \
+      iso_8859_15 | *.ISO*8859-15 | *.ISO8859-15)
         Tloc=latin1 ;;
       *.IBM-1047)
         Tloc=cp1047 ;;
+      *.KOI8-R)
+        T=-Tkoi8-r ;;
       *)
         case "$LESSCHARSET" in
           utf-8)
@@ -55,6 +59,8 @@
             Tloc=latin1 ;;
           cp1047)
             Tloc=cp1047 ;;
+          koi8-r)
+            T=-Tkoi8-r ;;
           *)
             Tloc=ascii ;;
         esac ;;
@@ -118,7 +124,7 @@
 fi
 
 case $T in
-  -Tascii | -Tlatin1 | -Tutf8 | -Tcp1047)
+  -Tascii | -Tlatin1 | -Tutf8 | -Tcp1047 | -Tkoi8-r)
     ;;
   *)
     # ignore other devices and use locale fallback
