--- ../../../crypto/openssh-4/sshd_config	2006-07-23 21:06:47.000000000 -0700
+++ sshd_config	2006-11-20 09:54:26.000000000 -0700
@@ -1,4 +1,5 @@
 #	$OpenBSD: sshd_config,v 1.74 2006/07/19 13:07:10 dtucker Exp $
+#	$DragonFly: src/secure/usr.sbin/sshd/Attic/sshd_config.no_obj.patch,v 1.7 2007/03/29 00:49:01 pavalos Exp $
 
 # This is the sshd server system-wide configuration file.  See
 # sshd_config(5) for more information.
@@ -10,8 +11,13 @@
 # possible, but leave them commented.  Uncommented options change a
 # default value.
 
+# Note that some of DragonFly's defaults differ from OpenBSD's, and
+# DragonFly has a few additional options.
+
+#VersionAddendum DragonFly-20070328
+
 #Port 22
-#Protocol 2,1
+#Protocol 2
 #AddressFamily any
 #ListenAddress 0.0.0.0
 #ListenAddress ::
@@ -19,7 +25,6 @@
 # HostKey for protocol version 1
 #HostKey /etc/ssh/ssh_host_key
 # HostKeys for protocol version 2
-#HostKey /etc/ssh/ssh_host_rsa_key
 #HostKey /etc/ssh/ssh_host_dsa_key
 
 # Lifetime and size of ephemeral version 1 server key
@@ -34,7 +39,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin yes
+#PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 
@@ -56,7 +61,7 @@
 #PasswordAuthentication yes
 #PermitEmptyPasswords no
 
-# Change to no to disable s/key passwords
+# Change to no to disable OPIE and PAM authentication
 #ChallengeResponseAuthentication yes
 
 # Kerberos options
@@ -82,7 +87,7 @@
 
 #AllowTcpForwarding yes
 #GatewayPorts no
-#X11Forwarding no
+#X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PrintMotd yes
@@ -99,6 +104,8 @@
 #MaxStartups 10
 #PermitTunnel no
 
+#XAuthLocation /usr/pkg/xorg/bin/xauth
+
 # no default banner path
 #Banner /some/path
 
