--- sshd_config	2007-09-09 15:58:55 -0700
+++ sshd_config	2007-09-09 16:27:40 -0700
@@ -1,4 +1,5 @@
 #	$OpenBSD: sshd_config,v 1.75 2007/03/19 01:01:29 djm Exp $
+#	$DragonFly: src/secure/usr.sbin/sshd/Attic/sshd_config.no_obj.patch,v 1.10 2007/09/09 23:32:43 corecode Exp $
 
 # This is the sshd server system-wide configuration file.  See
 # sshd_config(5) for more information.
@@ -10,6 +11,11 @@
 # possible, but leave them commented.  Uncommented options change a
 # default value.
 
+# Note that some of DragonFly's defaults differ from OpenBSD's, and
+# DragonFly has a few additional options.
+
+#VersionAddendum DragonFly-20070907
+
 #Port 22
 #AddressFamily any
 #ListenAddress 0.0.0.0
@@ -23,7 +29,6 @@
 # HostKey for protocol version 1
 #HostKey /etc/ssh/ssh_host_key
 # HostKeys for protocol version 2
-#HostKey /etc/ssh/ssh_host_rsa_key
 #HostKey /etc/ssh/ssh_host_dsa_key
 
 # Lifetime and size of ephemeral version 1 server key
@@ -38,7 +43,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin yes
+#PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 
@@ -60,7 +65,7 @@
 #PasswordAuthentication yes
 #PermitEmptyPasswords no
 
-# Change to no to disable s/key passwords
+# Change to no to disable OPIE and PAM authentication
 #ChallengeResponseAuthentication yes
 
 # Kerberos options
@@ -86,7 +91,7 @@
 
 #AllowTcpForwarding yes
 #GatewayPorts no
-#X11Forwarding no
+#X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PrintMotd yes
@@ -103,6 +108,8 @@
 #MaxStartups 10
 #PermitTunnel no
 
+#XAuthLocation /usr/pkg/bin/xauth
+
 # no default banner path
 #Banner /some/path
 
