$DragonFly: src/lib/libz/patches/zconf.h.patch,v 1.1 2004/11/12 18:10:42 joerg Exp $

--- zconf.h.orig	2004-11-12 18:56:36.000000000 +0100
+++ zconf.h	2004-11-12 18:56:46.000000000 +0100
@@ -5,8 +5,8 @@
 
 /* @(#) $Id$ */
 
-#ifndef ZCONF_H
-#define ZCONF_H
+#ifndef _ZCONF_H
+#define _ZCONF_H
 
 /*
  * If you *really* need a unique prefix for all types and library functions,
@@ -323,4 +323,4 @@
 #   pragma map(inflate_copyright,"INCOPY")
 #endif
 
-#endif /* ZCONF_H */
+#endif /* _ZCONF_H */
