$DragonFly: src/lib/libz/patches/zlib.h.patch,v 1.1 2004/11/12 18:10:42 joerg Exp $

--- zlib.h.orig	2004-10-23 15:49:54.000000000 +0200
+++ zlib.h	2004-11-12 18:57:57.000000000 +0100
@@ -28,8 +28,8 @@
   (zlib format), rfc1951.txt (deflate format) and rfc1952.txt (gzip format).
 */
 
-#ifndef ZLIB_H
-#define ZLIB_H
+#ifndef _ZLIB_H
+#define _ZLIB_H
 
 #include "zconf.h"
 
@@ -1197,4 +1197,4 @@
 }
 #endif
 
-#endif /* ZLIB_H */
+#endif /* _ZLIB_H */
